atoma-input-manager/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+use std::path::Path;
+
+use atoma_helpers::Firebase;
+use atoma_types::{AtomaInputMetadata, InputSource};
+use config::AtomaInputManagerConfig;
+use firebase::FirebaseInputManager;
+use thiserror::Error;
+use tokio::sync::mpsc;
+use tracing::{info, instrument};
+
+mod config;
+mod firebase;
+
+/// `AtomaInputManager` - manages different input sources
+///     requests, allowing for a flexible interaction between
+///     the user and the Atoma Network.
+///
+/// Current available options for storing the input on behalf
+/// of the user consists of Firebase.
+pub struct AtomaInputManager {
+    /// Firebase's input manager instance.
+    firebase_input_manager: FirebaseInputManager,
```

Comment:
The goal is to add support, over time, for different sources to interact with Atoma, right ?

Reply:
Yes.